packages/core/src/projections/augury-event-projection.class.ts

```diff
@@ -0,0 +1,18 @@
+import { AuguryEvent } from '../events';
+import { Projection } from './projection.interface';
+
+export abstract class AuguryEventProjection<Output> implements Projection<AuguryEvent, Output> {
+  public abstract process(event: AuguryEvent): boolean;
+
+  public collectResult(): Output | null {
+    const output = this.getOutput();
+
+    this.cleanup();
+
+    return output;
+  }
+
+  protected abstract getOutput(): Output | null;
```

Comment:
Perhaps the term `output` should renamed to reflect the `result` nomenclature for consistency? This way it's clear the `output` and `result` are related.